uspv/init.go

```diff
@@ -74,45 +75,46 @@ func (s *SPVCon) GetListOfNodes() ([]string, error) {
 }
 
 // DialNode receives a list of node ips and then tries to connect to them one by one.
-func (s *SPVCon) DialNode(listOfNodes []string) error {
+func (s *SPVCon) DialNode(listOfNodes []string) (net.Conn, error) {
+
 	// now have some IPs, go through and try to connect to one.
 	var err error
-	for i, ip := range listOfNodes {
+	var con net.Conn
+	for _, ip := range listOfNodes {
 		// try to connect to all nodes in this range
-		var conString, conMode string
 		// need to check whether conString is ipv4 or ipv6
+		var conString string
+		var conMode string
 		conString, conMode, err = s.parseRemoteNode(ip)
-		log.Printf("Attempting connection to node at %s\n",
-			conString)
+		if err != nil {
+			log.Printf("parse error for node (skipped): %s", err)
+			continue
+		}
 
+		log.Printf("Attempting connection to node at %s...", conString)
 		if s.ProxyURL != "" {
 			log.Printf("Attempting to connect via proxy %s", s.ProxyURL)
 			var d proxy.Dialer
 			d, err = proxy.SOCKS5("tcp", s.ProxyURL, nil, proxy.Direct)
 			if err != nil {
-				return err
+				return nil, err
 			}
 
-			s.con, err = d.Dial(conMode, conString)
+			con, err = d.Dial(conMode, conString)
 		} else {
-			s.con, err = net.Dial(conMode, conString)
+			con, err = net.Dial(conMode, conString)
 		}
 
 		if err != nil {
-			if i != len(listOfNodes)-1 {
```

Comment:
I guess this got mixed up during rebases.. https://github.com/mit-dci/lit/pull/270/commits/06417fdd2b4941cffa7686ae7b0234d3b50e7c3b doesn't contain this change